tensorflow/contrib/ignite/kernels/ignite_holder.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "ignite/ignite.h"
+#include "ignite/ignition.h"
+
+using namespace ignite;
+
+class IgniteHolder {
+public:
+    static IgniteHolder &Instance() {
+        static IgniteHolder s;
```

Comment:
Is `Ignite` trivially destructable? If not, use the following:

```c++
static IgniteHolder* s = new Ignite;
return *s;
```

Reply:
Can you please elaborate why this singleton implementation is not correct, cpp, unfortunately, is not my primary language :)

Reply:
It might be, but it also might trigger undefined behavior: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

If you can't state definitively that it is safe, please make the change I suggested above.

Reply:
Done.